dace/transformation/helpers.py

```diff
@@ -17,6 +18,307 @@
 from dace.memlet import Memlet
 
 
+def nest_sdfg_subgraph(sdfg: SDFG,
+                       subgraph: SubgraphView,
+                       start: Optional[SDFGState] = None,
+                       name: Optional[str] = None) -> SDFGState:
+
+    # Nest states
+    states = subgraph.nodes()
+    if len(states) > 1:
+
+        if start is not None:
+            source_node = start
+        else:
+            src_nodes = subgraph.source_nodes()
+            if len(src_nodes) != 1:
+                raise NotImplementedError
```

Comment:
probably better to raise a proper error here:
`raise ValueError('Ambiguous start state for nesting SDFG subgraph, please provide a single start state via the "start" argument')`